File: scala/src/main/scala/dojo/Diamond.scala
package dojo


class Diamond(val middleLetter: Char) {

  def rows: Seq[Seq[Char]] = ???

}

object Diamond {
  def print(middleLetter: Char): String = {
    val diamond = new Diamond(middleLetter)
    diamond.rows.map(_.mkString).mkString("\n")
  }

  def main(args: Array[String]): Unit = {
    if (args.size == 1) {
      val diamondLetter = args(0).charAt(0)
      System.out.print(print(diamondLetter))
    } else {
      System.out.println("please supply one argument: the char of the diamond middle")
    }
  }
}

File: java/src/main/java/codingdojo/Diamond.java
package codingdojo;

public class Diamond {
    private char middleLetter;

    public Diamond(char middleLetter) {
        this.middleLetter = middleLetter;
    }

    public String printDiamond() {
        return "A\n";
    }

    public static void main(String[] args) {
        if (args.length > 0) {
            System.out.println(new Diamond(args[0].charAt(0)).printDiamond());
        } else {
            System.out.println("please supply one argument: the char of the diamond middle");
        }
    }
}
